src/dpp/cluster.cpp

```diff
@@ -300,7 +300,7 @@ json error_response(const std::string& message, http_request_completion_t& rv)
 
 void cluster::post_rest(const std::string &endpoint, const std::string &major_parameters, const std::string &parameters, http_method method, const std::string &postdata, json_encode_t callback, const std::string &filename, const std::string &filecontent, const std::string &filemimetype) {
 	/* NOTE: This is not a memory leak! The request_queue will free the http_request once it reaches the end of its lifecycle */
-	rest->post_request(new http_request(endpoint + "/" + major_parameters, parameters, [endpoint, callback](http_request_completion_t rv) {
+	rest->post_request(new http_request(endpoint + (!major_parameters.empty() ? "/" : "") + major_parameters, parameters, [endpoint, callback](http_request_completion_t rv) {
```

Comment:
```diff
- (!major_parameters.empty() ? "/" : "") + major_parameters
+ (!major_parameters.empty() ? "/" + major_parameters : "")
```
Would be slightly better I think! Not super important though.
(Same for the other one below)

Reply:
hm. not a fan of longer ternary operators. they make it harder to read if you ask me